colcon-test/action.yaml

```diff
@@ -0,0 +1,119 @@
+  label:
+    description: ""
+    required: false
+    default: ".*" # Run all tests
+  token:
+    description: ""
+    required: false
+    default: ${{ github.token }}
+outputs:
+  coverage-reports:
+    description: ""
+    value: ${{ steps.test.outputs.coverage-reports }}
+
+runs:
+  using: composite
+  steps:
+    - name: Show target packages
+      run: |
+        echo "target packages: ${{ inputs.target-packages }}"
+      shell: bash
+
+    - name: Set up colcon-mixin
+      run: |
+        colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml || true
+        colcon mixin update default
+      shell: bash
+
+    - name: Install packages for coverage commands
+      run: |
+        sudo apt-get -yqq update
+        sudo apt-get -yqq install lcov python3-pip
+        pip3 install colcon-lcov-result colcon-coveragepy-result
+      shell: bash
+
+    - name: Clone dependency packages
+      if: ${{ inputs.build-depends-repos != ''}}
+      run: |
+        git config --local --unset-all http.https://github.com/.extraheader || true
+        git config --global url.https://x-access-token:${{ inputs.token }}@github.com.insteadOf 'https://github.com'
+        mkdir dependency_ws
+        vcs import dependency_ws < ${{ inputs.build-depends-repos }}
+      shell: bash
+
+    - name: Run rosdep install
+      run: |
+        sudo apt-get -yqq update
+        rosdep update
+        DEBIAN_FRONTEND=noninteractive rosdep install -yqq --from-paths . --ignore-src --rosdistro ${{ inputs.rosdistro }}
+      shell: bash
+
+    - name: Restore build files from cache
+      id: restore-build-files
+      uses: actions/cache@v2
+      with:
+        path: |
+          ./build
+          ./install
+        key: build-${{ github.sha }}
+
+    - name: Test
+      id: test
+      run: |
+        set -e
+        . /opt/ros/${{ inputs.rosdistro }}/setup.sh
+
+        colcon lcov-result --zero-counters \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --lcov-base lcov_${{ inputs.label }}
+
+        colcon lcov-result --initial \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --lcov-base lcov_${{ inputs.label }}
+
+        colcon test --event-handlers console_cohesion+ \
+          --mixin coverage-pytest \
+          --packages-above ${{ inputs.target-packages }} \
+          --ctest-args -L "${{ inputs.label }}" \
+          --return-code-on-test-failure
+
+        colcon lcov-result \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --lcov-base lcov_${{ inputs.label }} || true
+
+        colcon coveragepy-result \
+          --verbose \
+          --packages-above ${{ inputs.target-packages }} \
+          --coveragepy-base coveragepy_${{ inputs.label }} || true
+
+        coverage_files=()
+        lcov_coverage_file=lcov_${{ inputs.label }}/total_coverage.info
```

Comment:
Why didn't you use the `check-file-existence` action?
https://github.com/autowarefoundation/autoware-github-actions/tree/tier4/proposal/check-file-existence

Reply:
I modified the script ff6b0ee